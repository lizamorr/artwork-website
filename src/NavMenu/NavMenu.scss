.nav-menu {
  align-self: baseline;
  cursor: pointer;
  z-index: 10;
  padding: 0;
  margin: 0;
  opacity: 1 !important;

  .menu-icon--animated {
    border-radius: 8px;

    -webkit-animation: glowing 3000ms infinite;
    -moz-animation: glowing 3000ms infinite;
    -o-animation: glowing 3000ms infinite;
    animation: glowing 3000ms infinite;

    @-webkit-keyframes glowing {
      20% {
        -webkit-box-shadow: 0 0 10px #e5f4facc;
      }
      50% {
        -webkit-box-shadow: 0 0 40px #eeee;
      }
      100% {
        -webkit-box-shadow: 0 0 10px #eeee;
      }
    }

    @-moz-keyframes glowing {
      20% {
        -moz-box-shadow: 0 0 10px #e5f4facc;
      }
      50% {
        -moz-box-shadow: 0 0 40px #eeee;
      }
      100% {
        -moz-box-shadow: 0 0 10px #eeee;
      }
    }

    @-o-keyframes glowing {
      20% {
        box-shadow: 0 0 10px #e5f4facc;
      }
      50% {
        box-shadow: 0 0 40px #eeee;
      }
      100% {
        box-shadow: 0 0 10px #eeee;
      }
    }

    @keyframes glowing {
      20% {
        box-shadow: 0 0 10px #e5f4facc;
      }
      50% {
        box-shadow: 0 0 40px #eeee;
      }
      100% {
        box-shadow: 0 0 10px #eeee;
      }
    }
  }

  @media screen and (min-width: 767px) {
    .menu-icon--animated,
    .menu-icon,
    .close-icon {
      font-size: 45px;
    }
  }
  @media screen and (min-width: 1200px) {
    .menu-icon--animated,
    .menu-icon,
    .close-icon {
      font-size: 50px;
    }
  }
}
