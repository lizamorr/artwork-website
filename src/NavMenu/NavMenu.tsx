import React, { useState } from 'react';
import Menu from './Menu';
import MenuRoundedIcon from '@mui/icons-material/MenuRounded';
import CloseRoundedIcon from '@mui/icons-material/CloseRounded';

const NavMenu = () => {
  const [isOpen, setIsOpen] = useState(false);
  return (
    <div
      onClick={() => setIsOpen(!isOpen)}
      className="nav-menu"
      aria-label="Toggle menu"
    >
      {!isOpen ? (
        <MenuRoundedIcon className="menu-icon" sx={{ fontSize: 40 }} />
      ) : (
        <CloseRoundedIcon className="close-icon" sx={{ fontSize: 40 }} />
      )}
      {isOpen && <Menu />}
    </div>
  );
};

export default NavMenu;
