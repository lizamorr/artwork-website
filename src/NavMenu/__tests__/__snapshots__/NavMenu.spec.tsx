import NavMenu from '../../NavMenu';
import renderer from 'react-test-renderer';
import React from 'react';

describe('NavMenu Component', () => {
  it('should render correctly', () => {
    const component = renderer.create(<NavMenu />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
