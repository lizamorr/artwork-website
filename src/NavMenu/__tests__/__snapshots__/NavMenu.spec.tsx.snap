// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenu Component should render correctly 1`] = `
<div
  aria-label="Toggle menu"
  className="nav-menu"
  onClick={[Function]}
>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1ha5wk2-MuiSvgIcon-root{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;font-size:40px;}",
      }
    }
    data-emotion="css 1ha5wk2-MuiSvgIcon-root"
  />
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium menu-icon css-1ha5wk2-MuiSvgIcon-root"
    data-testid="MenuIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
    />
  </svg>
</div>
`;
